homeassistant/components/lamarzocco/coordinator.py

```diff
@@ -80,6 +87,43 @@ async def _async_init_client(self) -> None:
                 name="lm_websocket_task",
             )
 
+        # initialize Bluetooth
+        if self.config_entry.data.get(CONF_USE_BLUETOOTH, True):
+
+            def bluetooth_configured() -> bool:
+                return self.config_entry.data.get(
+                    CONF_MAC, ""
+                ) and self.config_entry.data.get(CONF_NAME, "")
+
+            if not bluetooth_configured():
+                for discovery_info in async_discovered_service_info(self.hass):
+                    name = str(discovery_info.name)
+                    if (name.startswith(tuple(BT_MODEL_NAMES))) and (
+                        name.split("_")[1] == self.config_entry.data[CONF_MACHINE]
+                    ):
+                        _LOGGER.debug(
+                            "Found Bluetooth device, configuring with Bluetooth"
+                        )
+                        # found a device, add MAC address to config entry
+                        new_data = self.config_entry.data.copy()
+                        new_data[CONF_MAC] = discovery_info.address
+                        new_data[CONF_NAME] = discovery_info.name
+                        self.hass.config_entries.async_update_entry(
+                            self.config_entry,
+                            data=new_data,
```

Comment:
You could instead unpack the existing entry with ** and append the new items here